archive/phar_without_stub.ksy

```diff
@@ -0,0 +1,219 @@
+meta:
+  id: phar_without_stub
+  title: PHP phar archive (without stub)
+  application: PHP
+  file-extension: phar
+  xref:
+    wikidata: Q1269709
+  license: CC0-1.0
+  ks-version: 0.9
+  imports:
+    - /serialization/php_serialized_value
+  endian: le
+doc: |
+  A phar (PHP archive) file. The phar format is a custom archive format from the PHP ecosystem that is used to package a complete PHP library or application into a single self-contained archive. All phar archives start with an executable PHP stub, which can be used to allow executing or including phar files as if they were regular PHP scripts. PHP 5.3 and later include the phar extension, which adds native support for reading and manipulating phar files.
+  
+  The phar format was originally developed as part of the PEAR library PHP_Archive, first released in 2005. Later, a native PHP extension named "phar" was developed, which was first released on PECL in 2007, and is included with PHP 5.3 and later. The phar extension has effectively superseded the PHP_Archive library, which has not been updated since 2010. The phar extension is also no longer released independently on PECL; it is now developed and released as part of PHP itself.
+  
+  Because of current limitations in Kaitai Struct (see kaitai-io/kaitai_struct#158 and kaitai-io/kaitai_struct#538), the executable PHP stub that precedes the rest of the archive is not handled by this spec. Before parsing a phar using this spec, the stub must be removed manually.
+  
+  A phar's stub is terminated by the special token `__HALT_COMPILER();` (which may be followed by at most one space, the PHP tag end `?>`, and an optional line terminator). The stub termination sequence is immediately followed by the remaining parts of the phar format, as described in this spec.
+  
+  The phar stub usually contains code that loads the phar and runs a contained PHP file, but this is not required. A minimal valid phar stub is `<?php __HALT_COMPILER();` - such a stub makes it impossible to execute the phar directly, but still allows loading or manipulating it using the phar extension.
+  
+  Note: The phar format does not specify any encoding for text fields (stub, alias name, and all file names), so these fields may contain arbitrary binary data. The actual text encoding used in a specific phar file usually depends on the application that created the phar, and on the standard encoding of the system on which the phar was created.
```

Comment:
Please [wrap these lines to have 80 columns limit](http://doc.kaitai.io/ksy_style_guide.html#doc). Using long lines causes problems with many pieces of our tooling (i.e. GitHub, some text editors, and, ultimately, these result in the source code generated with very long comments, which typically breaks target languages style guides).

Reply:
re. @KOLANICH: If you disagree with the style guide's length limit, can we discuss this in a separate issue? FWIW, I personally prefer soft wraps and no hard line limit, because it's compatible with all display sizes/layouts (unless you have to use an editor/tool that doesn't support soft wraps...), but if the style guide says 80 columns I'm going to follow that.

Reply:
@KOLANICH has once started a PR with changed to style guide: https://github.com/kaitai-io/kaitai_struct_doc/pull/7 — which I've mostly disagreed with.

We can have a discussion, but, unfortunately, I don't think there's much room to argue here. We're more or less tied to how the infrastructure around us works: i.e. how target languages work, how GitHub / text editors / similar tools work.

Just about *every* programming language I know insists on having a certain max line length. Even if programming language does not enforce it (like with C/C++), typically all the style guides that I know do. I haven't yet seen a single programming language so far that says "format all your code / documentation in a long single line".

And this is exactly what we're doing here: we're creating these docstrings to be inserted into target language source code. It's much easier to stick with the common practices of all other programming languages that to invent our own rules and then advocate for "proper" tools which support these.

Reply:
Yeah, I'm not suggesting that the official style should be changed. I have my reasons why I prefer no line breaks, but the project's code style takes precedence - it's more important to have a consistent code style at all (otherwise you have a mix of styles, and no matter which one you choose it will sometimes be wrong).

The Kaitai Struct code style is quite conservative with the 80 character limit, but at least this way it causes the least possible issues with other tools and code styles. Though to be exact, having an 80 character line limit in KSY files doesn't guarantee that the generated code will also fit into 80 columns - if the text is indented deeper in the generated code than in the KSY file, it will slightly exceed 80 characters. But I don't think we should try to fix that, because the only thing worse  
than a
small line length limit is text that was badly rewrapped into 80  
columns.